From f35ecf08892356e655c0e5f17fd01c149ca527c3 Mon Sep 17 00:00:00 2001
From: Ryan Zhang <ryan.zhang@nxp.com>
Date: Mon, 8 Dec 2025 15:07:35 +0800
Subject: [PATCH] fix: Initialize handle to prevent freeing uninitialized
 pointers.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In the default constructor, textures.envmap.sampler is initialized
to a non‑null garbage value (not VK_NULL_HANDLE). When timestamps
aren’t supported, load_texture_cubemap() doesn’t overwrite it, so
the destructor calls vkDestroySampler(...) with this invalid
handle, causing a crash.

Upstream-Status: Submitted [https://github.com/KhronosGroup/Vulkan-Samples/pull/1453]
Signed-off-by: Ryan Zhang <ryan.zhang@nxp.com>
---
 framework/api_vulkan_sample.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/framework/api_vulkan_sample.h b/framework/api_vulkan_sample.h
index 113be3e9..45208555 100644
--- a/framework/api_vulkan_sample.h
+++ b/framework/api_vulkan_sample.h
@@ -58,7 +58,7 @@ struct SwapchainBuffer
 struct Texture
 {
 	std::unique_ptr<vkb::sg::Image> image;
-	VkSampler                       sampler;
+	VkSampler                       sampler = VK_NULL_HANDLE;
 };
 
 /**
-- 
2.43.0

